src/Summoner/Template.hs

```diff
@@ -0,0 +1,528 @@
+{-# LANGUAGE QuasiQuotes         #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | This module contains functions for stack template creation.
+
+module Summoner.Template
+       ( createStackTemplate
+       ) where
+
+import Data.Semigroup ((<>))
+import Data.Text (Text)
+import NeatInterpolation (text)
+
+import Summoner.Default (defaultGHC, endLine)
+import Summoner.ProjectData (ProjectData (..))
+
+import qualified Data.Text as T
+
+----------------------------------------------------------------------------
+-- Stack File Creation
+----------------------------------------------------------------------------
+
+emptyIfNot :: Bool -> Text -> Text
+emptyIfNot p txt = if p then txt else ""
+
+-- | Creating template file to use in `stack new` command
+createStackTemplate :: ProjectData ->  Text
+createStackTemplate
+    ProjectData{..} = createCabalTop
+                   <> emptyIfNot isLib createCabalLib
+                   <> emptyIfNot isExe
+                                 ( createCabalExe
+                                 $ emptyIfNot isLib $ ", " <> repo )
+                   <> emptyIfNot test createCabalTest
+                   <> emptyIfNot bench
+                                 ( createCabalBenchmark
+                                 $ emptyIfNot isLib $ ", " <> repo )
+                   <> emptyIfNot github createCabalGit
+                   <> createCabalFiles
+                   <> readme
+                   <> emptyIfNot github gitignore
+                   <> emptyIfNot travis travisYml
+                   <> emptyIfNot appVey appVeyorYml
+                   <> emptyIfNot script scriptSh
+                   <> changelog
+                   <> createLicense
+                   <> createStackYamls testedVersions
+  where
+    -- all basic project information for `*.cabal` file
+    createCabalTop :: Text
+    createCabalTop =
+        [text|
+        {-# START_FILE ${repo}.cabal #-}
+        name:                $repo
+        version:             0.0.0
+        description:         $description
+        synopsis:            $description
+        homepage:            https://github.com/${owner}/${repo}
+        bug-reports:         https://github.com/${owner}/${repo}/issues
+        license:             $license
+        license-file:        LICENSE
+        author:              $nm
+        maintainer:          $email
+        copyright:           $year $nm
+        category:            $category
+        build-type:          Simple
+        extra-doc-files:     README.md
+        cabal-version:       >=1.24
```

Comment:
When I tested projects recently I noticed that `>=` shouldn't exist here. Just `1.24`.